server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -144,8 +144,8 @@ ParserContext createMultiFieldContext(ParserContext in) {
                 return new MultiFieldParserContext(in);
             }
 
-            ParserContext createDynamicTemplateFieldContext(ParserContext in) {
-                return new DynamicTemplateParserContext(in);
+            ParserContext createDynamicTemplateFieldContext() {
```

Comment:
Is this the only usage of the parser context function? should we change the function argument directly to create the dynamic parser context directly instead of going through a parser context to create another one?

Reply:
It's also used when validating the templates where we *do* want to throw an error rather than issuing warnings, so we need to be able to distinguish between the two cases - hence the extra step.

Reply:
ok :)